esc: clarify RPM/command direction meaning

tpwrules opened the pull request:
Reverse should only be used for reverse thrust, not simply a motor that happens to be spinning counter-clockwise.

Some ESC vendors get the RPM feedback backwards when configured in reverse direction, so flight stacks need to be aware; not sure how wide-spread the problem is. I have confirmed that latest AM32 using DroneCAN behaves correctly here: reversing the motor direction in its configuration does not reverse the RPM sign.

I had an additional comment about how an ESC which is running in reverse in response to a supported negative bi-directional command must report a negative RPM but AM32 does not do this. I am not sure where the limitation is but IMO this is a bug, and probably a more wide-spread one.

I further wonder how we should define the motor angle in the extended status. I presume it should be that increasing angles are clockwise? So it would swap if you reversed direction in the ESC, and always match the physical angle. But I am not sure what's best here and don't own any ESCs with angle sensors. We do need to add a note that the percentages are absolute.

tridge commented on uavcan/equipment/esc/1030.RawCommand.uavcan:
```diff
@@ -1,7 +1,9 @@
+# Raw ESC command normalized into [-8192, 8191]. The ESC should normalize the setpoint into its effective input range.
 # Non-zero setpoint value below minimum should be interpreted as min valid setpoint for the given motor.
+# Positive values indicate forward rotation, negative values indicate reverse rotation, and zero indicates no rotation.
+# Forward rotation shall be the direction causing forward thrust. For example, a typical quadcopter always commands
```
remove "Forward rotation shall be the direction causing forward thrust". "forward" is a problem, eg. copters have no "forward" thrust

tridge commented on uavcan/equipment/esc/1031.RPMCommand.uavcan:
```diff
@@ -2,7 +2,9 @@
 # for example, given a ESC that operates in the range 100 to 10000 RPM, a setpoint of 1 RPM will be clamped to 100 RPM.
-# Negative values indicate reverse rotation.
+# Positive values indicate forward rotation, negative values indicate reverse rotation, and zero indicates no rotation.
+# Forward rotation shall be the direction causing forward thrust. For example, a typical quadcopter always sets
```
same as above

tpwrules commented:
Removed the comment about thrust and added recommendations for reversible ESCs.

Hopefully we can get AM32 to do negative RPMs.

I will also ask the original authors about the meaning of the motor angle and document it in a separate PR.
